app/src/main/java/org/mozilla/reference/browser/compose/ComposeFragment.kt

```diff
@@ -0,0 +1,47 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.reference.browser.compose
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+import androidx.compose.ui.platform.ViewCompositionStrategy
+import androidx.fragment.app.Fragment
+
+/**
+ * Base class for fragments that use Compose UI and have a 1:1 relationship with a Composable, i.e
+ * Fragments that do not have [View]s and [ComposeView]s defined in the XML layout.
+ *
+ * The ViewCompositionStrategy is set to [ViewCompositionStrategy.DisposeOnLifecycleDestroyed],
+ * meaning that the [ComposeView] will be disposed when the Fragment's view is destroyed.
+ *
+ * Read more about [ViewCompositionStrategy] here:
+ * https://medium.com/androiddevelopers/viewcompositionstrategy-demystefied-276427152f34
+ * https://developer.android.com/develop/ui/compose/migrate/interoperability-apis/compose-in-views#composition-strategy
+ */
+abstract class ComposeFragment : Fragment() {
```

Comment:
As a separate note: It would be nice to upstream and use this from `components.compose.base`